File: CPP/Leetcode/682-baseball-game/682-baseball-game.cpp
class Solution {
public:
    int calPoints(vector<string>& ops) {
        vector<int> v;
        
        for(int i=0;i<ops.size();i++)
        {
            string s=ops[i];
            char c=s[0];
            int f=0;
            if(c!='D' && c!='C' && c!='+')
            {
                int k=stoi(s);
                v.push_back(k);
            }
            if(c=='C')
            {
                v.pop_back();
            }
            if(c=='D')
            {
                int k=v[v.size()-1]*2;
                v.push_back(k);
            }
            if(c=='+')
            {
                int k=v[v.size()-1]+v[v.size()-2];
                v.push_back(k);
            }
        }
        int sum=0;
        for(auto it:v)
        {
            sum+=it;
        }
        return sum;
    }
};

File: Some Leetcode and GFG Solutions/682-baseball-game/682-baseball-game.py
class Solution:
    def calPoints(self, ops: List[str]) -> int:
        x=[]
        for i in ops:
            if i=="+":
                x.append(x[-1]+x[-2])
            elif i=="C":
                x.pop()
            elif i=="D":
                x.append(x[-1]*2)
                
            else:
                x.append(int(i))
        return sum(x)